src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -425,32 +428,20 @@ public void DeleteAttachmentBeforeRevert(DocumentsOperationContext context, Lazy
             }
         }
 
-        public void RevisionAttachments(DocumentsOperationContext context, Slice lowerId, Slice changeVector)
+        public void RevisionAttachments(DocumentsOperationContext context, BlittableJsonReaderObject document, Slice lowerId, Slice changeVector)
         {
-            using (GetAttachmentPrefix(context, lowerId.Content.Ptr, lowerId.Size, AttachmentType.Document, Slices.Empty, out Slice prefixSlice))
+            var attachmentsFromDocumentMetadata = GetAttachmentsFromDocumentMetadata(document)
+                .Select(attachment => JsonDeserializationClient.AttachmentName(attachment)).ToList();
```

Comment:
Can we avoid linq calls here? They allocate

Reply:
done